jujuclient/file.go

```diff
@@ -547,6 +543,27 @@ func updateModels(
 	return nil
 }
 
+// SetModels implements ModelUpdater.
+func (s *store) SetModels(controllerName string, models map[string]ModelDetails) error {
+	if err := ValidateControllerName(controllerName); err != nil {
+		return errors.Trace(err)
+	}
+
```

Comment:
The validattion happens below inside updateModels. 
My thinking was that since I am updating a collection, it's best to check validity closer to the write. I am happy to move it out though - it's causing me puzzlement withing the update method :D I am pretty sure I've left a to-do because of it.